File: order-generator/OrderGenerator/Models/Item.cs
using Newtonsoft.Json;

namespace OrderGenerator.Models
{
    public class Item
    {
        [JsonProperty("id")]
        public required string Id { get; set; }

        [JsonProperty("order_id")]
        public required string OrderId { get; set; }

        [JsonProperty("description")]
        public required string Description { get; set; }

        [JsonProperty("price")]
        public float Price { get; set; }
    }
}

File: common/models/Item.py
from pydantic import BaseModel

class Item(BaseModel):
    id: str
    order_id: str
    description: str
    price: float
    
    @classmethod
    def _convert_to_real_type_(cls, data):
        data_type = data.get("type")

        if data_type is None:
            raise ValueError("Missing 'type' in Item")

        sub = cls._subtypes_.get(data_type)

        if sub is None:
            raise TypeError(f"Unsupport sub-type: {data_type}")

        return sub(**data)
    
    @classmethod
    def parse_obj(cls, obj):
        return cls._convert_to_real_type_(obj)